### PR TITLE
Fix/optimize map helper

### PR DESCRIPTION
**Pull request by btamas**
Test: `npm run test`

### CODE REVIEW

File: src/helpers/map.js [hunk 1]
```diff
@@ -73,18 +73,9 @@ export default {
...
-            if (sections) {
-                _.forEach(sections, function(section) {
-                    result[section.id] = section;
-                });
-            }
-        });
-        return result;
+        const parts = this.getParts(map) || {};
+
+        return Object.values(parts).reduce((result, part) => Object.assign(result, part.sections), {});
```
Comment (ferenckiss89) [inline src/helpers/map.js]: can it happen that we don't have sections?

File: src/helpers/map.js [hunk 2]
```diff
@@ -93,25 +84,21 @@ export default {
...
-                        _.forEach(items, function(item) {
-                            if (item.active) {
-                                result = item;
-                            }
-                        });
+        const parts = this.getParts(map) || {};
+
+        for (let part of Object.values(parts)) {
+            const sections = part.sections || {};
+            for (let section of Object.values(sections)) {
```
Comment (ferenckiss89) [inline src/helpers/map.js]: does it make sense to continue if `sections` is an empty object?

File: src/helpers/map.js (hunk 1 above)
Comment (btamas) [reply src/helpers/map.js]: Yes it can

File: src/helpers/map.js (hunk 2 above)
Comment (btamas) [reply src/helpers/map.js]: It won't continue, because empty object has no values. But I will add a condition instead.

**Pull request closed without merging**